### ISSUE TITLE
Kafka external access does not work with TLS enabled

### ISSUE THREAD

**Issue by mpmatti** [bitnami/charts#8875]
**Which chart**:

kafka-14.1.0

**Describe the bug**

We have configured external access to kafka (type loadBalancer) with client protocol SASL_TLS enabled. Access from clients does not work, it gives the following error:

```
ERROR [Consumer clientId=consumer-console-consumer-68764-1, groupId=console-consumer-68764] Connection to node 2147483647 (x.x.x.x/x.x.x.x:9094) failed authentication due to: SSL handshake failed (org.apache.kafka.clients.NetworkClient)
ERROR Error processing message, terminating consumer process:  (kafka.tools.ConsoleConsumer$)
org.apache.kafka.common.errors.SslAuthenticationException: SSL handshake failed
Caused by: javax.net.ssl.SSLHandshakeException: No subject alternative names matching IP address x.x.x.x found

```

**To Reproduce**

1. generate TLS cert for kafka server with three hostnames in SAN (DNS records already exist)
Add the key, cert and CA cert in a kubernetes secret object 'kafka-pem'. 

2. install kafka with the following values.yaml:

```yaml
kafka:
  replicaCount: 3 
  persistence: 
    enabled: true
  readinessProbe:
    initialDelaySeconds: 60
  livenessProbe:
    initialDelaySeconds: 60     
  externalAccess:
    enabled: true
    service:
      type: LoadBalancer
      port: 9094
      loadBalancerIPs: 
        - x.x.x.x
        - y.y.y.y
        - z.z.z.z
    
  auth:
    clientProtocol: sasl_tls
    interBrokerProtocol: plaintext

    tls:    
      type: pem
      existingSecret: kafka-pem
      
    sasl:
      jaas:
        existingSecret: kafka-secrets    
        interBrokerUser: brokeruser
        zookeeperUser: zookeeperuser
        clientUsers:
          - kafka-user

  allowPlaintextListener: true
  deleteTopicEnable: true
  autoCreateTopicsEnable: true 

  zookeeper:
    allowAnonymousLogin: false
    replicaCount: 3
    auth:
     enabled: true
     existingSecret: kafka-zookeeper-secrets
     clientUser: zookeeperuser
     serverUsers: zookeeperuser

```

3. set up a client using trusting the same CA cert

4. run kafka shell commands in client, e.g.
`kafka-console-producer.sh --producer.config /config/client-secret --bootstrap-server kafka-0.mydomain.com:9094 --topic test`

**Expected behavior**

The clients should be able to connect to the server, create topics, etc.

**Further investigations**

kafka server.properties file contains only the IP address of the kafka node in advertised listeners:
```
I have no name!@kafka-0:/$ cat /opt/bitnami/kafka/config/server.properties |grep -i advertised
advertised.listeners=INTERNAL://kafka-0.kafka-headless.kafka.svc.cluster.local:9093,CLIENT://kafka-0.kafka-headless.kafka.svc.cluster.local:9092,EXTERNAL://x.x.x.x:9094

```
Which causes the problem, as it advertises only the IP address to the client, and that does not match the SAN hostnames in the server certificate. It seems the chart does not have any configuration option to provide kafka with the correct hostnames used for external listener.

We also tested by adding the external host IPs to the certificate SAN list, and it worked fine. However, defining IP addresses in SAN is really not optimal, as in the end we'd like to use a wildcard certificate. Is there any way - using the chart parameters - to do this properly. Can we make kafka to advertise hostnames instead of IP addresses?

**Comment by carrodher** [comment]
Are you using this chart as subchart? Please note the first `kafka:` level in the values.yaml shouldn't be needed, [here](https://github.com/bitnami/charts/tree/master/bitnami/kafka#enable-security-for-kafka-and-zookeeper) you can find more info about configuring TLS. Note the parameters are not preceded by any `kafka.` level

**Comment by mpmatti** [comment]
Yes, we are using it as subchart. But that is not the issue here.

**Comment by fmulero** [comment]
Hi @mpmatti. We've tested the configuration shared in latest version (15) and we are not able to reproduce this issue. Please be aware that `auth.tls.existingSecret`was deprecated in version 14 and now you have to use `auth.tls.existingSecrets` with a list of certificates, one per replica. Could you give it a try and share the results? Thanks a lot

**Comment by mpmatti** [comment]
Hi @fmulero,
Thank you for your response. In fact, I updated to the chart/kafka version 15.0.2/3.1.0, and switched to using auth.tls.existingSecrets accordingly. The issue still remains. The problem is that kafka advertises the broker IP address, not the hostname, and as the client tries to connect via IP, it does not match the server TLS certificate, as we don't have IP addresses added to Subject Alternative Name field in cert. As you see from this line, EXTERNAL listener is only configured as an IP address:
https://github.com/bitnami/charts/blob/980410291ace3c6fbe202f12de874e9216330139/bitnami/kafka/templates/scripts-configmap.yaml#L115
I think there should be a way to tell kafka to advertise hostnames instead of IP addresses.

**Comment by fmulero** [comment]
Hi @mpmatti 

We've just merged the changes to allow using names in external listeners. Two new parameters have been added:
* `externalAccess.service.loadBalancerNames`: It takes precedence over load balancer IPs, if loadBalancerNames is not defined then IP addresss will be used for external access in broker configuration. If this parameter is set, `externalAccess.service.loadBalancerIPs` is not mandatory. Length must be the same as replicaCount
* `externalAccess.service.loadBalancerAnnotations`: Array of load balancer annotations for external services. Useful to apply different annotations to each external load balancer. For example if external-dns is used to manage a DNS entry for the load balancer. Length must be the same as replicaCount.

The new chart version will be published in the repository during today
